### PR TITLE
add new CaloConfig table and support to L1T CondFormats

### PR DESCRIPTION
**Pull request by mulhearn**
This adds a new table to CondFormats/L1Trigger to contain firmware version information.  This is a pure addition, which does not yet effect any of the existing code.  This table will be needed to wean L1T off of customizations!

### CODE REVIEW

**Comment by mulhearn** [comment]
+1

**Comment by diguida** [comment]
@cmsbuild please notice the change

File: CondFormats/L1TObjects/interface/CaloConfig.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+namespace l1t {
+
+
+  class CaloConfig {
+
+  public:
+
+    enum { Version = 1 };
+
+    CaloConfig() { version_=Version; }
```
Comment (diguida) [inline CondFormats/L1TObjects/interface/CaloConfig.h]: @mulhearn here I can see an implicit conversion from non-strongly typed `enum` to `unsigned int`. Can this be made explicit?

**Comment by mulhearn** [comment]
@diguida yes will fix and update this evening...  sorry no time all day to fix it yet.

**Comment by mulhearn** [comment]
@cmsbuild please retest

**Comment by mulhearn** [comment]
Oh christ, its unsigned... fixing.  Should only take a minute.

**Comment by mulhearn** [comment]
@cmsbuild please retest

**Comment by mulhearn** [comment]
+1

**Comment by diguida** [comment]
+1

**Comment by ggovi** [comment]
+1

**Pull request merged**